server/log-collector/pkg/services/logcollector/server.go

```diff
@@ -1055,22 +1066,29 @@ func (s *Server) isLogCollectionRunning(ctx context.Context, runUID, project str
 	}
 }
 
-// getChunkSuze returns the minimum between the request size, buffer size and the remaining size to read
-func (s *Server) getChunkSize(requestSize, endSize, currentOffset int64) int64 {
+// getChunkSuze returns the size of the chunk to read from the log file
+func (s *Server) getChunkSize(
+	initialOffset,
+	totalLogsSize,
+	endSize int64) int64 {
 
-	chunkSize := int64(s.getLogsBufferSizeBytes)
-
-	// if the request size is smaller than the buffer size, use the request size
-	if requestSize > 0 && requestSize < chunkSize {
-		chunkSize = requestSize
+	// we read it all, chunk size is 0
+	if totalLogsSize+initialOffset >= endSize {
+		return 0
 	}
 
-	// if the remaining size is smaller than the buffer size, use the remaining size
-	if endSize-currentOffset < chunkSize {
-		chunkSize = endSize - currentOffset
+	// if we didn't read anything yet, assume read the initial offset
+	if totalLogsSize == 0 {
+		totalLogsSize += initialOffset
```

Comment:
same here, relates to the comment above

Reply:
resolves it as the answer relates to https://github.com/mlrun/mlrun/pull/4531#discussion_r1383374543